### PR TITLE
Chore(Fail-Steps): Updated fail-step and error messages for Azure experiments

### PR DESCRIPTION
**Pull request by avaakash**
Signed-off-by: Akash Shrivastava <akash.shrivastava@harness.io>

**What this PR does / why we need it**:
This PR updates the fail-step and error messages for the following Azure experiment
1. azure-instance-stop
2. azure-disk-loss

Additionally,

Fixes issue with empty IDs provided

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
-   [ ] Fixes #<issue number>
-   [ ] PR messages has document related information
-   [ ] Labelled this PR & related issue with `breaking-changes` tag
-   [ ] PR messages has breaking changes related information
-   [ ] Labelled this PR & related issue with `requires-upgrade` tag
-   [ ] PR messages has upgrade related information
-   [ ] Commit has unit tests
-   [ ] Commit has integration tests
-   [ ] E2E run Required for the changes

### CODE REVIEW

File: pkg/cloud/azure/disk/disk-operations.go [hunk 1]
```diff
@@ -55,23 +65,26 @@ func DetachDisks(subscriptionID, resourceGroup, azureInstanceName, scaleSet stri
...
-		}
 		vmClient.Authorizer = authorizer
 
 		// Fetch the vm instance
 		vm, err := vmClient.Get(context.TODO(), resourceGroup, azureInstanceName, compute.InstanceViewTypes("instanceView"))
 		if err != nil {
-			return errors.Errorf("fail get instance, err: %v", err)
+			return cerrors.Error{
+				ErrorCode: cerrors.ErrorTypeChaosInject,
+				Reason:    fmt.Sprintf("failed get instance: %v", err),
```
Comment (ispeakc0de) [inline pkg/cloud/azure/disk/disk-operations.go:85]:
```suggestion
				Reason:    fmt.Sprintf("failed to get instance: %v", err),
```

File: pkg/cloud/azure/disk/disk-operations.go [hunk 2]
```diff
@@ -164,11 +193,15 @@ func WaitForDiskToAttach(experimentsDetails *types.ExperimentDetails, diskName s
...
-				return errors.Errorf("failed to get the disk status, err: %v", err)
+				return stacktrace.Propagate(err, "failed to get the disk status")
 			}
 			if diskState != "Attached" {
-				log.Infof("[Status]: Disk %v is not yet attached, state: %v", diskName, diskState)
-				return errors.Errorf("Disk %v is not yet attached, state: %v", diskName, diskState)
+				log.Infof("[Status]: Disk %v is not yet attached, current state: %v", diskName, diskState)
+				return cerrors.Error{
+					ErrorCode: cerrors.ErrorTypeChaosRevert,
+					Reason:    fmt.Sprintf("Disk is not yet attached, current state: %v", diskState),
```
Comment (ispeakc0de) [inline pkg/cloud/azure/disk/disk-operations.go:202]:
```suggestion
					Reason:    fmt.Sprintf("Disk is not attached within timeout, disk state: %s", diskState),
```

File: pkg/cloud/azure/disk/disk-operations.go [hunk 3]
```diff
@@ -183,11 +216,15 @@ func WaitForDiskToDetach(experimentsDetails *types.ExperimentDetails, diskName s
...
 			if err != nil {
-				return errors.Errorf("failed to get the disk status, err: %v", err)
+				return stacktrace.Propagate(err, "failed to get the disk status")
 			}
 			if diskState != "Unattached" {
 				log.Infof("[Status]: Disk %v is not yet detached, state: %v", diskName, diskState)
-				return errors.Errorf("Disk %v is not yet detached, state: %v", diskName, diskState)
+				return cerrors.Error{
+					ErrorCode: cerrors.ErrorTypeChaosInject,
+					Reason:    fmt.Sprintf("Disk is not yet detached, current state: %v", diskState),
```
Comment (ispeakc0de) [inline pkg/cloud/azure/disk/disk-operations.go:225]:
```suggestion
					Reason:    fmt.Sprintf("Disk is not detached within timeout, disk state: %s", diskState),
```

File: pkg/cloud/azure/disk/disk-status.go [hunk 4]
```diff
@@ -2,60 +2,69 @@ package azure
...
 		vmClient.Authorizer = authorizer
 
 		// Fetch the vm instance
 		scaleSetName, vmId := common.GetScaleSetNameAndInstanceId(azureInstanceName)
 		vm, err := vmClient.Get(context.TODO(), resourceGroup, scaleSetName, vmId, compute.InstanceViewTypes("instanceView"))
 		if err != nil {
-			return nil, errors.Errorf("fail get instance, err: %v", err)
+			return nil, cerrors.Error{
+				ErrorCode: cerrors.ErrorTypeTargetSelection,
+				Reason:    fmt.Sprintf("failed get instance: %v", err),
```
Comment (ispeakc0de) [inline pkg/cloud/azure/disk/disk-status.go:40]:
```suggestion
				Reason:    fmt.Sprintf("failed to get instance: %v", err),
```

File: pkg/cloud/azure/disk/disk-status.go [hunk 5]
```diff
@@ -2,60 +2,69 @@ package azure
...
-	if err != nil {
-		return nil, errors.Errorf("fail to setup authorization, err: %v", err)
-	}
-	vmClient.Authorizer = authorizer
+		// Fetch the vm instance
+		vm, err := vmClient.Get(context.TODO(), resourceGroup, azureInstanceName, compute.InstanceViewTypes("instanceView"))
+		if err != nil {
+			return nil, cerrors.Error{
+				ErrorCode: cerrors.ErrorTypeTargetSelection,
+				Reason:    fmt.Sprintf("failed get instance: %v", err),
```
Comment (ispeakc0de) [inline pkg/cloud/azure/disk/disk-status.go:59]:
```suggestion
				Reason:    fmt.Sprintf("failed to get instance: %v", err),
```

File: pkg/cloud/azure/disk/disk-status.go [hunk 6]
```diff
@@ -86,20 +102,37 @@ func CheckVirtualDiskWithInstance(subscriptionID, virtualDiskNames, resourceGrou
...
+			return cerrors.Error{
+				ErrorCode: cerrors.ErrorTypeStatusChecks,
+				Reason:    fmt.Sprintf("failed to get disk: %v", err),
+				Target:    fmt.Sprintf("{Azure Disk Name: %v, Resource Group: %v}", diskName, resourceGroup)}
 		}
 		if disk.ManagedBy == nil {
-			return errors.Errorf("disk %v not attached to any instance", diskName)
+			return cerrors.Error{
+				ErrorCode: cerrors.ErrorTypeStatusChecks,
+				Reason:    "disk not attached to any instance",
```
Comment (ispeakc0de) [inline pkg/cloud/azure/disk/disk-status.go:134]:
```suggestion
				Reason:    "disk is not attached to any instance",
```

File: pkg/cloud/azure/instance/instance-operations.go [hunk 7]
```diff
@@ -117,16 +150,20 @@ func WaitForAzureComputeDown(timeout, delay int, scaleSet, subscriptionID, resou
...
 			}
 			if err != nil {
-				return errors.Errorf("failed to get the instance status")
+				return stacktrace.Propagate(err, "failed to get the instance status")
 			}
 			if instanceState != "VM stopped" {
-				return errors.Errorf("instance is not yet in stopped state")
+				return cerrors.Error{
+					ErrorCode: cerrors.ErrorTypeChaosInject,
+					Reason:    "instance is not yet in stopped state",
```
Comment (ispeakc0de) [inline pkg/cloud/azure/instance/instance-operations.go:158]:
```suggestion
					Reason:    "instance is not in stopped within timeout",
```

File: pkg/cloud/azure/instance/instance-operations.go [hunk 8]
```diff
@@ -146,10 +183,14 @@ func WaitForAzureComputeUp(timeout, delay int, scaleSet, subscriptionID, resourc
...
 			}
 			if err != nil {
-				return errors.Errorf("failed to get instance status")
+				return stacktrace.Propagate(err, "failed to get instance status")
 			}
 			if instanceState != "VM running" {
-				return errors.Errorf("instance is not yet in running state")
+				return cerrors.Error{
+					ErrorCode: cerrors.ErrorTypeChaosRevert,
+					Reason:    "instance is not yet in running state",
```
Comment (ispeakc0de) [inline pkg/cloud/azure/instance/instance-operations.go:191]:
```suggestion
					Reason:    "instance is not in running state within timeout",
```

**Review by ispeakc0de — APPROVED** [review]

File: pkg/cloud/azure/instance/instance-operations.go (hunk 7 above)
Comment (avaakash) [reply pkg/cloud/azure/instance/instance-operations.go]: `instance did not stop within timeout` would be the correct sentence IMO.

**Pull request merged**